### ISSUE TITLE
cannot handle theories

### ISSUE THREAD

**Issue by jwaldmann** [jwaldmann/haskell-tpdb#18]
.. in the XTC parser. - Input looks like this:
```
<signature>
<funcsym><name>plus</name><arity>2</arity><theory>AC</theory></funcsym>
<funcsym><name>0</name><arity>0</arity></funcsym>
<funcsym><name>s</name><arity>1</arity></funcsym>
</signature>
```
source code: https://github.com/jwaldmann/haskell-tpdb/blob/master/src/TPDB/XTC/Read.hs#L86

to handle this properly, need to extend the type of RS.signature. Currrently, it is list of symbols,
but actually it should be Map from Symbol to Theory annotation.

the "old" format has this in a different presentation, cf. https://www.lri.fr/~marche/tpdb/format.html

**Comment by jwaldmann** [comment]
We could just ignore the annotation? This is dangerous, because it changes semantics.

We can put theory information in `Problem` (instead of `RS`), currently
```
data Problem v s =
     Problem { type_ :: Type
             , trs :: TRS v s
             , strategy :: Maybe Strategy
             -- , metainformation :: Metainformation
             , startterm :: Maybe Startterm
             , attributes :: Attributes
             }
```

**Comment by jwaldmann** [comment]
sort-of fixed by commit aa2d12d605617ecfd735e4ea8ad9fa2e5bf641eb
